src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1681,14 +1705,269 @@ GenTree* Lowering::LowerHWIntrinsicCmpOp(GenTreeHWIntrinsic* node, genTreeOps cm
         }
         else
         {
+            assert(simdSize == 16);
+
             // TODO-Review: LowerHWIntrinsicCC resets the id again, so why is this needed?
             node->ChangeHWIntrinsicId(NI_SSE41_TestZ);
             LowerHWIntrinsicCC(node, NI_SSE41_PTEST, cmpCnd);
         }
 
-        return node->gtNext;
+        return LowerNode(node);
```

Comment:
this won't lead to a recursion, right?

Reply:
Right, we've explicitly changed from a comparison op to a `Test` op, so its really just handling the containment support.

We do this type of `return LowerNode(node)` quite frequently in the HWIntrinsic logic to avoid code duplication, to avoid mistakes around `->gtNext`, and to ensure containment/other optimizations still happens for newly introduced nodes